sql/plan/insert.go

```diff
@@ -449,10 +520,16 @@ func (ii InsertInto) DebugString() string {
 	return pr.String()
 }
 
-func validateNullability(dstSchema sql.Schema, row sql.Row) error {
-	for i, col := range dstSchema {
-		if !col.Nullable && row[i] == nil {
-			return ErrInsertIntoNonNullableProvidedNull.New(col.Name)
+func (i *insertIter) validateNullability(dstSchema sql.Schema, row sql.Row) error {
+	for count, col := range dstSchema {
+		if !col.Nullable && row[count] == nil {
+			// In the case of an IGNORE we set the nil value to a default and add a warning
+			if i.ignore {
+				row[count] = col.Type.Zero()
+				_ = i.handleErrorAndQuitOrIgnore(sql.ErrInsertIntoNonNullableProvidedNull.New(col.Name)) // will always return nil
```

Comment:
It looks like you're using this to add a warning? I'd write a comment stating as such. Also maybe break `handleErrorAndQuitOrIgnore` up and have a function that only warns on ignorable errors. It isn't immediately clear what calling a function that has _quit or ignore_ and ignoring the returned variable is for.

Maybe name it `warnOnIgnorableError` or something.